pkg/antctl/command_definition.go

```diff
@@ -320,6 +328,159 @@ func respTransformer(obj interface{}) (interface{}, error) {
 	return target, nil
 }
 
+// getTableHeaders get the headers of the output table.
+func (cd *commandDefinition) getTableHeaders(target interface{}, tab *table.Writer) ([]string, error) {
+	args := make([]string, 0)
+	headers, ok := target.(map[string]interface{})
+	if !ok {
+		return nil, errors.New("converting types error in getTableHeaders")
+	}
+
+	// ensure that "name" column is the first column if it exists
+	// sort other columns in alphabet order in default
+	hasNameHeader := false
+	nameHeader := ""
+	tabConfigs := make([]table.ColumnConfig, 0)
+	for header := range headers {
+		if strings.ToLower(header) == "name" {
+			hasNameHeader = true
+			nameHeader = header
+			continue
+		}
+		args = append(args, header)
+	}
+	sort.Strings(args)
+	if hasNameHeader {
+		args = append([]string{nameHeader}, args...)
+	}
+
+	tabHeaders := table.Row{}
+	// set the column configs
+	for _, arg := range args {
+		colName := strings.ToUpper(arg)
+		tabHeaders = append(tabHeaders, colName)
+		tabConfig := table.ColumnConfig{
+			Name:     colName,
+			WidthMax: maxTableOutputColumnLength,
+		}
+		tabConfigs = append(tabConfigs, tabConfig)
+	}
+	(*tab).AppendHeader(tabHeaders)
+	(*tab).SetColumnConfigs(tabConfigs)
+
+	return args, nil
+}
+
+func (cd *commandDefinition) handleListTableElement(list []interface{}, arg string) (string, error) {
+	if arg == "rules" {
+		return strconv.Itoa(len(list)), nil
+	}
```

Comment:
Are you still working on moving these specific logic to transform packages? It won't be maintainable in long term by adding  special handling of each column of each resource here.
I'm thinking there could be an Interface defined for table header and rows conversion and make each `Response` struct implement it. Then they can decide how to convert their own fields to cells of a row. The generic code just needs to check if the struct implement the Interface and calls its method to get headers and rows if yes.

Reply:
> Are you still working on moving these specific logic to transform packages? It won't be maintainable in long term by adding special handling of each column of each resource here.
> I'm thinking there could be an Interface defined for table header and rows conversion and make each `Response` struct implement it. Then they can decide how to convert their own fields to cells of a row. The generic code just needs to check if the struct implement the Interface and calls its method to get headers and rows if yes.


This solution is reasonable yet costly, for that we have to rewrite all responses. 

My solution is based on the previous committed code. 
For a list:
1. For structs, we find out if there is a field called "PrintName", and if so, we use it as the table element.
2. For non-structs, we find out if it is a string or integer, if so, we directly use it as the table element.
3. If the 2 conditions are not satisfied at the same time, we use the length of list as the default table element.

Reply:
That cost is for long term maintenance and will be needed anyway, otherwise the rules you list will get longer each time a resource or a column that doesn't meet the current requirement is added, we shouldn't let generic function know much about concrete type and do many assumption. It should only build simple, generic contract with concrete types.